### PR TITLE
Allow filtering exported metrics in OperatorConfig

### PR DESCRIPTION
**Pull request by fabxc**
* Add an OperatorConfig section to set the --export.match flag.
* Fix CRD YAML generation concatenation
* Ensure that e2e tests are truly e2e by default (verify writes to GCM)

### CODE REVIEW

File: pkg/operator/e2e/context.go [hunk 1]
```diff
@@ -79,6 +80,9 @@ func newTestContext(t *testing.T) *testContext {
 		namespace:      namespace,
 		kubeClient:     kubeClient,
 		operatorClient: operatorClient,
+		// Pick a random port to avoid conflicts with other simultaneous tests in the cluster
+		// as the collector runs on the host network.
+		collectorPort: 1025 + rand.Int31n(65536-1025),
```
Comment (pintohutch) [inline pkg/operator/e2e/context.go:85]: What does moving this here do?

File: pkg/operator/e2e/main_test.go [hunk 2]
```diff
@@ -407,23 +415,42 @@ func testCollectorDeployed(ctx context.Context, t *testContext) {
...
+				"--storage.tsdb.no-lockfile",
+				"--storage.tsdb.retention.time=30m",
+				"--storage.tsdb.wal-compression",
+				"--storage.tsdb.min-block-duration=10m",
+				"--storage.tsdb.max-block-duration=10m",
+				fmt.Sprintf("--web.listen-address=:%d", t.collectorPort),
+				"--web.enable-lifecycle",
+				"--web.route-prefix=/",
+				fmt.Sprintf("--export.label.project-id=%s", projectID),
+				fmt.Sprintf("--export.label.cluster=%s", cluster),
```
Comment (pintohutch) [inline pkg/operator/e2e/main_test.go:439]: What about `fmt.Sprintf("--export.label.location=%s", location)`?

Comment (fabxc) [reply pkg/operator/e2e/main_test.go:439]: Oh, I see we did add a flag – I did run directly on my machine to verify GCM.
Will take care of it.

**Comment by fabxc** [comment]
Running GCM validation with `make test` doesn't make sense as the provided projectID etc. aren't real.

**Review by pintohutch — APPROVED** [review]

**Pull request merged**